xdsl/interactive/app.py

```diff
@@ -165,14 +165,19 @@ class InputApp(App[None]):
     (i.e. is the Output TextArea).
     """
     pass_pipeline = reactive(tuple[tuple[type[ModulePass], PipelinePassSpec], ...])
-    """Reactive variable that saves the list of selected passes."""
+    """Reactive variable that saves the selected passes."""
 
     condense_mode = reactive(False, always_update=True)
     """Reactive boolean."""
     available_pass_list = reactive(tuple[type[ModulePass], ...])
     """
-    Reactive variable that saves the list of passes that have an effect on
-    current_module.
+    Reactive variable that saves the passes that have an effect on current_module.
+    """
+    current_possible_rewrites = reactive(
+        tuple[tuple[int, tuple[str, str, RewritePattern], ModuleOp]]
+    )
+    """
+    Reactive variable that saves the rewrite patterns that have an effect on current_module.
     """
```

Comment:
Conceptually, the new rewrites are not any different to the passes, both because they end up in the pass pipeline in the top right quarter, but also because they'll all be represented as a pass in the `pass_pipeline`. I feel like it would be worth spending a bit of time now to make them more uniform at the input stage. I would propose something like this:
```
    condense_mode = reactive(False, always_update=True)
    show_individual_rewrites_mode = reactive(False, always_update=True)
    available_pass_list = reactive(tuple[tuple[type[ModulePass], PipelinePassSpec|None], ...])
```
Where the available pass list contains both the passes (with None instead of a PipelinePassSpec, opening the window if pass parameters are required) and the individual rewrites, where the rewrite info is stored in the PipelinePassSpec in the same element of the tuple.

Reply:
It would probably make sense to make a separate PR to prepare for this change, and rebase this one on top of that

Reply:
I can do a separate PR in future to add a button that does a "rewrite mode" so to speak.